typhon-core/src/lib.rs

```diff
@@ -59,10 +60,19 @@ const _: () = {
         fn get() -> &'static Self {
             CELL.get().expect("Settings were not initialized")
         }
-        fn init(settings: Self) {
+        fn init(password: &String) {
+            let password = Box::leak(Box::new(password.clone()));
+            let password = PasswordHash::new(password).expect("Unable to parse the password hash");
+            let settings = Self { password };
             CELL.set(settings)
                 .expect("Settings were already initalized")
         }
+        fn verify_password(&self, password: &[u8]) -> bool {
+            use argon2::{Argon2, PasswordVerifier};
+            Argon2::default()
+                .verify_password(password, &self.password)
+                .is_ok()
+        }
```

Comment:
in my opinion `verify_password`  should not be an associated item of `Settings`, but a standalone function, e.g.:
```suggestion
        fn verify_admin_password(password: &[u8]) -> bool {
            use argon2::{Argon2, PasswordVerifier};
            Argon2::default()
                .verify_password(password, &Settings::get().password)
                .is_ok()
        }
```